### PR TITLE
Separate ActionFormat into ROS1 and ROS2 versions

### PR DESCRIPTION
**Pull request by schmerl**
For the moment, functionality is duplicated across ROS1 and ROS2 implementation. Related to #420

### CODE REVIEW

File: src/roswire/common/action.py [hunk 1]
```diff
@@ -25,26 +27,26 @@ class ActionFormat:
...
+    definition: str = attr.ib()
+    goal: MF = attr.ib()
+    feedback: Optional[MF] = attr.ib()
+    result: Optional[MF] = attr.ib()
 
-    @staticmethod
+    @classmethod
     def from_file(
-        package: str, filename: str, files: dockerblade.FileSystem
+            cls, package: str, filename: str, files: dockerblade.FileSystem
```
Comment (ChrisTimperley) [inline src/roswire/common/action.py:49]: Excess whitespace

File: src/roswire/common/action.py [hunk 2]
```diff
@@ -79,52 +82,14 @@ def from_string(package: str, name: str, s: str) -> "ActionFormat":
...
-        return ActionFormat(package, name, s, goal, feed, res)
-
-    @staticmethod
+        ...
+
+    @classmethod
+    @abstractmethod
     def from_dict(
-        d: Dict[str, Any], *, package: Optional[str] = None
+            cls, d: Dict[str, Any], *, package: Optional[str] = None
```
Comment (ChrisTimperley) [inline src/roswire/common/action.py:90]: excess whitespace

File: test/test_format.py [hunk 3]
```diff
@@ -9,9 +9,8 @@
     Field,
     MsgFormat,
     SrvFormat,
-    ActionFormat,
 )
-from roswire.ros1 import ROS1PackageDatabase, ROS1FormatDatabase
+from roswire.ros1 import ROS1ActionFormat, ROS1PackageDatabase, ROS1FormatDatabase
```
Comment (ChrisTimperley) [inline test/test_format.py:13]: prefer relative import

**Review by ChrisTimperley — CHANGES_REQUESTED** [review]
A few minor stylistic points, but otherwise this looks good 👍🏻

File: test/test_format.py (hunk 3 above)
Comment (schmerl) [reply test/test_format.py:13]: I think because this is a test case, we should do the `roswire.`, which is consistent with the other test cases.

**Pull request merged**